### ISSUE TITLE
RODARE Version

### ISSUE THREAD

**Issue by ax3l** [alpaka-group/llama#11]
At HZDR, go to publication database and start a "software publication".

https://www.hzdr.de/db/!WebAppl.Publ_Input.Data?pNid=108&pRedirect=%21Publications

It will provide us with a publication-ID with which we can create a [`.rodare.json`](https://github.com/openPMD/openPMD-api/blob/dev/.rodare.json) with which we can auto-generate DOIs for release tags.

**Comment by bernhardmgruber** [comment]
Do we need RODARE, when we can have Zenodo?

**Comment by psychocoderHPC** [comment]
Only zenodoo or redare. Important is to have a publication ID.

**Comment by bernhardmgruber** [comment]
Great, for Zenodoo we track this issue with #114!

**Comment by ax3l** [comment]
@bernhardmgruber: authors at HZDR, like many institutions, need to get formal approval for their scientific publications, which includes a first SW release. This is also a good process to go through for scientific output tracking/reporting and liability reasons. At the end of this approval process you get an *HZDR* publication ID.
@juckel streamlined this process through Rodare - we just had first versions of LLAMA in place before this service came online.

Please just quickly sync with @bussmann which service to use for llama.

**Comment by bussmann** [comment]
I'd prefer RODARE.

**Comment by bernhardmgruber** [comment]
@ax3l LLAMA has been released on Github a while back already by @theZiz. I would therefore argue that the public SW release already happened a long time ago. So attaching a DOI to that release is not a publication IMO.

As for the choice between RODARE and Zenodo: I picked Zenodo becaues it is also what alpaka uses. Furthermore Zenodo automatically integrates with Github, so a DOI is automatically created when a release is tagged on Github. That makes a lot of sense. Do you know if RODARE has the same integration? Is it possible to have LLAMA on Zenodo and RODARE with the same DOI?

@bussmann If you want LLAMA on RODARE, why is alpaka not on RODARE (and on zenodo instead)? I think all projects in the alpaka ecosystem should use the same publication platform.

**Comment by juckel** [comment]
Since RODARE is basically a Zenodo clone, the same GitHub (but also GitLab) integration is available. Both Zenodo and RODARE allow external DOIs. However, a DOI can only be resolved to one platform not both.

**Comment by psychocoderHPC** [comment]
note: alpaka was teleased long time before RODARE was available. Zenodo and RODARE has a version history, therefore changing the platform for a already existing released software will break the release history.
IMO it is not important to have all software in the alpaka group based on the same DOI provider.

Am 30. Oktober 2020 10:57:39 MEZ schrieb Bernhard Manfred Gruber <notifications@github.com>:

**Comment by psychocoderHPC** [comment]
@bernhardmgruber is llama 0.2.0 already released via zenodo with a DOI?

**Comment by bernhardmgruber** [comment]
@psychocoderHPC I think neither LLAMA 0.1.0 or LLAMA 0.2.0 is released on Zenodo or RODARE. So we can still choose. I created the .zenodo.json already as part of the last release, but it is not in use because I do not have the rights to enable the Zenodo integration (see: #114).

@juckel Thank you for the clarification!

**Comment by psychocoderHPC** [comment]
So a switch to RODARE is still possibleif required.

**Comment by bernhardmgruber** [comment]
Yes, we can still choose RODARE. And if the integration is as seemless as it is described on Zenodo, I have no preference!

**Comment by bussmann** [comment]
Hi. I have decided to be agnostic on this issue and let Bernhard decide as he has to live with this decision.
PRO RODARE: Hosted and maintained at HZDR, does a lot of things very nicely and can be extended by us if we need it.
PRO ZENODO: Well known, CERN uses it, the rest of alpaka uses it, maybe makes cross-referencing easier, maybe not.

**Comment by bernhardmgruber** [comment]
I talked to my supervisors at CERN and none of them had a preference either. One additional argument for Zenodo could be visibility, since the platform is around for a longer time.

Anyway, @bussmann and me performed a coin toss each and Zenodo won. I enabled the Github integration on Zenodo now for LLAMA.

Thanks to @ax3l for triggering the discusison and @juckel for the additional information!
